File: web/src/plugin/activityGuide/api/guide.js
import service from '@/utils/request'

export const createGuide = (data) => {
  return service({
    url: '/activityGuide/createGuide',
    method: 'post',
    data
  })
}

export const deleteGuide = (params) => {
  return service({
    url: '/activityGuide/deleteGuide',
    method: 'delete',
    params
  })
}

export const deleteGuideByIds = (params) => {
  return service({
    url: '/activityGuide/deleteGuideByIds',
    method: 'delete',
    params
  })
}

export const updateGuide = (data) => {
  return service({
    url: '/activityGuide/updateGuide',
    method: 'put',
    data
  })
}

export const findGuide = (params) => {
  return service({
    url: '/activityGuide/findGuide',
    method: 'get',
    params
  })
}

export const getGuideList = (params) => {
  return service({
    url: '/activityGuide/getGuideList',
    method: 'get',
    params
  })
}

File: server/plugin/activityGuide/router/guide.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/flipped-aurora/gin-vue-admin/server/plugin/activityGuide/api"
	"github.com/gin-gonic/gin"
)

var Guide = new(guide)

type guide struct{}

func (r *guide) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
	{
		group := private.Group("activityGuide").Use(middleware.OperationRecord())
		group.POST("createGuide", api.Guide.CreateGuide)
		group.DELETE("deleteGuide", api.Guide.DeleteGuide)
		group.DELETE("deleteGuideByIds", api.Guide.DeleteGuideByIds)
		group.PUT("updateGuide", api.Guide.UpdateGuide)
	}
	{
		group := private.Group("activityGuide")
		group.GET("findGuide", api.Guide.FindGuide)
		group.GET("getGuideList", api.Guide.GetGuideList)
	}
}
